dev-docs/RFCs/v8.1/gpu-accelerated-geo-fencing.md

```diff
@@ -0,0 +1,153 @@
+
+### Existing Research Work
+
+Following is the list of papers I was able to find and explored :
+
+1. Speeding up Large-Scale Point-in-Polygon Test Based Spatial Join on GPUs ([Paper](http://www-cs.ccny.cuny.edu/~jzhang/papers/bigspaital_cr.pdf))
+
+2. A cell-based point-in-polygon algorithm suitable for large sets of points ([Paper](https://www.sciencedirect.com/science/article/pii/S0098300401000371?via%3Dihub))
+
+3. GPU Rasterization for Real-Time Spatial Aggregation over Arbitrary Polygons ([Paper](http://www.vldb.org/pvldb/vol11/p352-zacharatou.pdf))
+
+4. A Simple and fast hardware-accelerated point-in-polygon test ([Paper](https://pdfs.semanticscholar.org/04a4/7b99d57cdf81bbd58a8dc21c2d15ac528855.pdf))
+
+All of above work uses more advanced Graphics APIs (such as OpenGL Compute Shaders, CUDA, SSBO, etc) and or not supported either by WebGL1 or WebGL2. For deck.gl, we have to implement this with in WebGL1/WebGL2 limitations.
+
+## Proposed Approach
+
+A texture based filtering can be implemented to perform PIP test on GPU. The idea is, user provided polygon or polygons are triangulated and rendered to a offline frame buffer. During actual rendering step it is used as a texture to perform PIP test. The texture size can be configured to control precession/speed. Bigger texture size has better precession while smaller texture size will have better speed.
+
+### PolygonTexture
+
+Axis aligned bounding box of polygons is first computed. A Framebuffer is created that has same aspect reatio as bounding box and it is cleared with a custom clear color (can be black). User provided polygons are triangulated and their positions are normalized to bounding box and are rendered to above offline Framebuffer using a custom color (we can just use one of RGB channels).
+
+### Filtering
+
+When rendering user provided data, for each data element, its positions is transformed into same space that above texture is generated and it position (XY) is normalized to a [0, 1] range and then we first perform trivial check agains bounding box (for range [0, 1]) to clip agains bounding box and then perform texture sample from above texture. The result gives whether the object is in the polygon or not.
+
+## Problem Classifications and proposed API
+
+### Filtering points
+
+We can perform a point in polygon test of each object and discard the rendering if test fails. One are more layer can be marked to generate the filter texture and one or more layers can be marked to use this texture as a mask for filtering. In layer's vertex shader, a texture sample is performed to determine if the object is inside or outside.
+
+Entire process of filtering can be divided into two steps, 1. Generation of filter texture and 2. performing actual filtering.
+
+* API : Following new components will be added
+
+#### GeoFilterEffect
```

Comment:
This does not have to be geospatial. Suggest avoid using the term `geo`. `MaskingEffect`?

Reply:
Yeah, will remove references to `geo`, given generated texture can be anything.